### PR TITLE
Bonnie 158 create test tool

### PR DESCRIPTION
**Pull request by alexanderelliott121**
Creates a test framework for qrda export and a selection of fixtures.

### CODE REVIEW

**Comment by edeyoung** [comment]
looks like there might be an issue with the config for running the tests in travis

File: test/functional/qrda_export_test.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+class RegistrationsControllerTest < ActionController::TestCase
+
+  include Devise::TestHelpers
+  
+  
+  setup do
+  end
+  
+  test "QRDA Export Test" do
```
Comment (edeyoung) [inline test/functional/qrda_export_test.rb:19]: add a high level comment here - iterating over the qrda test fixtures. compares output qrda with supplied qrda (or something along those lines)

File: test/functional/qrda_export_test.rb [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+        user = User.first
+        patient = Record.first
+        measure = Measure.first
+
+        associate_user_with_measures(user, [measure])
+        associate_user_with_patients(user, [patient])
+        associate_measure_with_patients(measure, [patient])
+
+        HealthDataStandards::SVS::ValueSet.all.each do |vs|
```
Comment (edeyoung) [inline test/functional/qrda_export_test.rb:38]: add comment for why you need to do this step

File: test/functional/qrda_export_test.rb [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        end_time = Time.new(Time.zone.at(APP_CONFIG['measure_period_start']).year, 12, 31)
+        
+        qrda_exporter = HealthDataStandards::Export::Cat1.new 'r3_1'
+        ret_file= qrda_exporter.export(patient, [measure], start_time, end_time, nil, 'r3_1')
+        ret_xml = Nokogiri::XML(ret_file)
+        
+        expected_file = Dir.glob(File.join("test", "fixtures", "qrda", test_dir, "*.xml"))[0]
+        expected_xml = Nokogiri::XML(File.read(expected_file))
+
+        tags_to_remove = ["ClinicalDocument id", "ClinicalDocument effectiveTime", "ClinicalDocument time"]
```
Comment (edeyoung) [inline test/functional/qrda_export_test.rb:55]: add comment explaining why these tags should be removed

File: test/functional/qrda_export_test.rb [hunk 4]
```diff
@@ -0,0 +1,70 @@
+require 'test_helper'
```
Comment (edeyoung) [inline test/functional/qrda_export_test.rb:1]: consider adding a comment to the top of this about how the directories should be formed. For example, there should be one fixture (or XML) file in each leaf directory.

**Review by edeyoung — CHANGES_REQUESTED** [review]

**Comment by edeyoung** [comment]
@alexanderelliott121 is test/functional/qrda_export_test.rb the only file with logic in it?

**Comment by edeyoung** [comment]
@mayerm94 @holmesie i'm interested in your thoughts. Alex and I had a long conversation about the directory structure for the tests. I think what he did makes a lot of sense but it shifts away from how this was done in other areas of the code.It allows you to simply iterate over the directory structures, which is really nice. I'm interested in what you guys think.

**Comment by alexanderelliott121** [comment]
@edeyoung 
Yes, qrda_export_test.rb is the only file with logic in it.

**Comment by alexanderelliott121** [comment]
Note, this is dependant on the hds changes before it will pass travis.

File: test/functional/qrda_export_test.rb (hunk 2 above)
Comment (alexanderelliott121) [reply test/functional/qrda_export_test.rb:38]: So, that's something that I put in when I was trying to fix the bson id problem and never removed.
It doesn't hurt, but it doesn't help either.

**Comment by edeyoung** [comment]
This change has been incorporated in this pull request: https://github.com/projecttacoma/bonnie/pull/653

**Comment by edeyoung** [comment]
Scratch my previous comment. the tool for generating fixtures is in pull request https://github.com/projecttacoma/bonnie/pull/653 but the new fixtures and tests for them is not.

**Review by edeyoung — CHANGES_REQUESTED** [review]
this likely needs some significant revisions to sit on top of the measure_history_rebase branch

**Comment by edeyoung** [comment]
@alexanderelliott121 is there anything from this pull request that we should pull over? If not, should we close it (given the reset test generation code you completed)?

**Comment by edeyoung** [comment]
closing. obsolete

**Pull request closed without merging**